operator/internal/manifests/openshift/var.go

```diff
@@ -37,6 +37,14 @@ func clusterRoleName(opts Options) string {
 	return opts.BuildOpts.GatewayName
 }
 
+func monitoringRoleName(stackName string) string {
+	return fmt.Sprintf("%s-metrics-reader", stackName)
```

Comment:
As we have surmised by now this change affects Prometheus' service discovery more than the actual scrape. Is `metrics-discovery` a better name?